src/Agent.cs

```diff
@@ -26,6 +26,14 @@ public class Agent
         private readonly Collector _collector;
         private static readonly ILog _log = LogManager.GetCurrentClassLogger();
 
+        public int MessageCount
+        {
+            get
+            {
+                return _collector.MessageCount;
```

Comment:
C# delegates are different from this.  I think this retains maximum clarity.

Honestly, I'd rather not expose this at all, and just use reflection to get this value in the test, but that's not where we are.